### ISSUE TITLE
Improving quadtree architecture

### ISSUE THREAD

**Issue by Jeremy-Gaillard** [iTowns/itowns#40]
A list of issues I stumbled upon while reading the code:
- SchemeTile seems to be barely used. Is it really necessary ?
- Can we remove node.wait and replace it by a test on the children's loaded attribute ? Or are there cases where a child is fully loaded but wait is still needed ?
- A child node shouldn't change the attribute of its parent (EllipsoidTileMesh.checkOrtho())

Secondly, I think we should define an interface for a generic hierarchical data structure (quadtree and octree being specific implementation of it). I think only the subdivide(node) function is needed.

The same remark applies to the tile composing the structure, but I think the Node class already has all the necessary attributes.

**Comment by gchoqueux** [comment]
- SchemeTile semble être peu utilisé. Is it really necessary ? -> yes, there will be others SchemesTile
- node.wait may be redundant
- A child node shouldn't change the attribute of its parent -> effectively

it's necessary to define an interface for a generic hierarchical data structure

**Comment by Jeremy-Gaillard** [comment]
While working on #43, I encountered problems with the quadtree initialisation.
In iTowns, to be able to add correctly a Layer (here, the quadtree) to the ThreeJS scene, the layer needs the root(s) of the 3D data to exist ([1](https://github.com/iTowns/itowns2/blob/master/src/Scene/Scene.js#L191), [2](https://github.com/iTowns/itowns2/blob/master/src/Scene/Layer.js#L47)).
Theoretically, a quadtree has no data at its initialisation (it needs to be downloaded first).
In the current (and, I guess, merely temporary) implementation, [the data is forcefully loaded in the constructor](https://github.com/iTowns/itowns2/blob/master/src/Scene/Quadtree.js#L31).
This is, of course, not an optimal solution. We need a way to create a scene root at the creation of the object.
[I tried making a workaround](https://github.com/iTowns/itowns2/blob/multiple_providers/src/Scene/Quadtree.js), but I am not satisfied. It basically adds an empty NodeMesh as the root of all the tiles.
~~Firstly, it is still buggy: the camera panning is broken (didn't find the source of the problem. Maybe @gchoqueux has an idea?)~~ EDIT: fixed. Secondly, it requires us to include the NodeMesh class in the Quadtree class, thus mixing data structure and rendering structure.
I thought about creating a root node of the type of the quadtree tiles (e.g. EllipsoidTileMesh), but this means that every Tile class will need a default constructor that creates an empty node.

In short: we need a way to initialise our layers without data while still adding them to the ThreeJS scene.

**Comment by gchoqueux** [comment]
@Jeremy-Gaillard this note is still relevant?

**Comment by Jeremy-Gaillard** [comment]
Part of this issue is still relevant. When creating a new quadtree (or any kind of data structure, the problem is not caused only by the quadtree implementation), level-0 tiles have to be loaded or the data will never be integrated to the scene. In some cases, this behaviour is not adequate. For example, a grid structure can't be implemented properly.

**Comment by gchoqueux** [comment]
Issue is still relevant?
